QuizApp/App/UseCase/LoginUseCase.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+protocol LoginUseCaseProtocol {
+
+    func loginUser(_ password: String, username: String) async throws -> LoginResponseModel
+
+}
+
+final class LoginUseCase: LoginUseCaseProtocol {
+
+    private let networkClient: NetworkClienProtocol
+
+    init(networkClient: NetworkClienProtocol) {
+        self.networkClient = networkClient
+    }
+
+    func loginUser(_ password: String, username: String) async throws -> LoginResponseModel {
+        guard let url = URL(string: Endpoint(type: .login).path) else {
```

Comment:
Building URL requests should be the job of LoginNetworkClient.
Also, please add DataSource layer in between UC and Network.